pytorch_lightning/core/lightning.py

```diff
@@ -1998,3 +2001,17 @@ def get_from_queue(self, queue: torch.multiprocessing.SimpleQueue) -> None:
         self.trainer.callback_metrics.update(
             apply_to_collection(callback_metrics, np.ndarray, lambda x: torch.tensor(x))
         )
+
+    @contextmanager
+    def prevent_trainer_and_dataloaders_deepcopy(self) -> None:
+        self._prevent_trainer_and_dataloaders_deepcopy = True
+        yield
+        self._prevent_trainer_and_dataloaders_deepcopy = False
```

Comment:
if this is a pubic method now, we would have to document it.
but doesn't it make more sense as a protected context manager?

Reply:
Yes, let's make it private.